cmd/ttn-lw-cli/commands/end_devices_split.go

```diff
@@ -34,20 +35,36 @@ var (
 	setEndDeviceToJS   = ttnpb.AllowedFieldMaskPathsForRPC["/ttn.lorawan.v3.JsEndDeviceRegistry/Set"]
 )
 
+func nonImplicitPaths(paths ...string) []string {
+	nonImplicitPaths := make([]string, 0, len(paths))
+	for _, path := range paths {
+		if path == "ids" || strings.HasPrefix(path, "ids.") {
```

Comment:
These should only be the primary identifiers - `ids.device_id` and `ids.application_ids`
`ids.dev_addr`, `ids.join_eui` and `ids.dev_eui` should not be implicit - we need to have a way to determine if those are specified in the request or not.

Reply:
We decided weeks ago that `ids` are implicitly part of the field mask. This just applies that decision.

Reply:
But this is just incorrect decision. Since we're going to validate EUIs and DevAddr(and possibly other identifiers there), all requests not containing valid primary identifiers, EUIs and DevAddr will fail validation. That's not what we want.
Also how would the users be able to change EUIs(something we want to allow later) or even the DevAddr?

Reply:
Afaik this has nothing to do with validation. The diff here is intended to not pass implicit fields as they are included by the concerning component (i.e. implicit).